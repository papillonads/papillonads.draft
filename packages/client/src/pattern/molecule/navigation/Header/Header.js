import React from 'react'
import { node } from 'prop-types'

const style = {
  position: 'sticky',
  top: 0,
  left: 0,
  right: 0,
  zIndex: 101,
}

const Header = ({ children }) => (
  <div className="bg-gray-dark" style={style}>
    <div className="main-nav d-flex flex-justify-between px-3 pl-md-4 pr-md-4 py-3 box-shadow bg-gray-dark Details js-details-container">
      <div className="d-flex flex-self-center flex-lg-auto mr-lg-2">
        <a
          href="https://github.com/mts/papillon-explorer"
          className="border-gray-dark no-underline text-white"
          aria-label="Papillon Explorer"
        >
          <svg
            height="28"
            aria-label="github-logo"
            className="octicon octicon-mark-github"
            viewBox="0 0 16 16"
            version="1.1"
            width="28"
            role="img"
          >
            <path
              fillRule="evenodd"
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
            />
          </svg>
        </a>
        <div className="flex-self-center Details-content--shown">
          <div className="dropdown js-menu-container js-select-menu">
            <h1 className="h4 text-normal pl-3 mt-1">
              <button
                className="btn-link text-white link-white no-underline js-menu-target"
                type="button"
                aria-haspopup="true"
                aria-expanded="false"
              >
                {children}
              </button>
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
)

Header.defaultProps = {
  children: null,
}

Header.propTypes = {
  children: node,
}

export default Header
