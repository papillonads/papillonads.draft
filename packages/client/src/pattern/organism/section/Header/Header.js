import React from 'react'
import { shape, string } from 'prop-types'
import cn from 'classnames'
import styles from './Header.scss'
import NavigationBar from '../../../molecule/header/NavigationBar/NavigationBar'
import SearchBar from '../../../molecule/header/SearchBar/SearchBar'

const Header = ({ className }) => {
  const headerClassList = cn(className, styles.header)

  return (
    <div className={headerClassList}>
      <NavigationBar className={styles.navigationBar} />
      <SearchBar className={styles.searchBar} />
    </div>
  )
}

Header.defaultProps = {
  className: '',
}

Header.propTypes = shape({
  className: string,
}).isRequired

export default Header
