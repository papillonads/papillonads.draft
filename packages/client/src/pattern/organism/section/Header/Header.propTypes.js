import { shape } from 'prop-types'
import { navigationBarPropTypes } from '../../../molecule/header/NavigationBar/NavigationBar.propTypes'
import { searchBarPropTypes } from '../../../molecule/header/SearchBar/SearchBar.propTypes'

export const headerPropTypes = shape({
  navigationBar: navigationBarPropTypes,
  searchBar: searchBarPropTypes,
}).isRequired
