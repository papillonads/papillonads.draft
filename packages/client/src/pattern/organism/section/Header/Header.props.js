import { string, shape } from 'prop-types'
import { navigationBarPropTypes } from '../../../molecule/header/NavigationBar/NavigationBar.props'
import { searchBarPropTypes } from '../../../molecule/header/SearchBar/SearchBar.propTypes'

export const headerDefaultProps = {
  className: '',
}

export const headerPropTypes = {
  className: string,
  data: shape({
    navigationBar: navigationBarPropTypes.data,
    searchBar: searchBarPropTypes,
  }).isRequired,
}
