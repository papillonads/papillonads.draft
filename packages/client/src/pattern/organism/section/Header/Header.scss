.header {
  width: 100%;
  margin-right: auto;
  margin-left: auto;

  @media (min-width: 992px) {
    width: 90%;
  }

  @media (min-width: 1280px) {
    width: 60%;
  }

  &__navigationBar {
  }

  &__searchBar {
  }
}
