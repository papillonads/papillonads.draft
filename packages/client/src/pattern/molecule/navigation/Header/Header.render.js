import React from 'react'
import Header from './Header'

const headerDefault = <Header>some text</Header>

export default headerDefault
