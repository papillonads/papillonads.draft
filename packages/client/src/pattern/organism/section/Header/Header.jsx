import React from 'react'
import cn from 'classnames'
import cs from './Header.scss'
import NavigationBar from '../../../molecule/header/NavigationBar'
import SearchBar from '../../../molecule/header/SearchBar'
import { headerDefaultProps, headerPropTypes } from './Header.props'

const Header = ({ className, data }) => (
  <div className={cn(className, cs.header)}>
    <NavigationBar data={data.navigationBar} />
    <SearchBar data={data.searchBar} />
  </div>
)

Header.defaultProps = headerDefaultProps

Header.propTypes = headerPropTypes

export default Header
