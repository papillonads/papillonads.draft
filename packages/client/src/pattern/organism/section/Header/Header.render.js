import React from 'react'
import Header from './Header'
import homePage from '../../../../../../server/src/datasources/page/homePage'

const defaultRender = <Header data={homePage.header} />

export { defaultRender }
