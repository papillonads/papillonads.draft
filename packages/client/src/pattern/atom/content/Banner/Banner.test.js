import { topBannerRender, marketingBannerRender } from './Header.render'

describe('<Banner />', () => {
  describe('Snaphot', () => {
    test('must match topBannerRender', () => {
      expect(global.renderToJSON(topBannerRender)).toMatchSnapshot()
    })

    test('must match marketingBannerRender', () => {
      expect(global.renderToJSON(marketingBannerRender)).toMatchSnapshot()
    })
  })
})
