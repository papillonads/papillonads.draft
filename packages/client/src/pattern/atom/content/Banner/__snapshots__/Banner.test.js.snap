// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> Snaphot must match marketingBannerRender 1`] = `
<div
  className="banner"
  style={
    Object {
      "height": "",
      "maxWidth": "",
    }
  }
>
  <iframe
    src="https://s0.2mdn.net/ads/richmedia/studio/pv2/60764566/20190205052314854/index.html"
    title="banner"
  />
</div>
`;

exports[`<Banner /> Snaphot must match topBannerRender 1`] = `
<div
  className="banner"
  style={
    Object {
      "height": "10em",
      "maxWidth": "730px",
    }
  }
>
  <iframe
    src="https://s0.2mdn.net/8054340/1535538205200/728x90.html"
    title="banner"
  />
</div>
`;
